src/distilabel/utils/dataset.py

```diff
@@ -51,3 +60,215 @@ def load_task_from_disk(path: Path) -> "Task":
     with open(task_path, "rb") as f:
         task = pickle.loads(f.read())
     return task
+
+
+def _binarize_dataset(
+    dataset: "CustomDataset",
+    seed: int = None,
+    strategy: BinarizationStrategies = "random",
+    keep_ties: bool = True,
+    **kwargs: Any,
+) -> "CustomDataset":
+    """Binarizes a distilabel dataset.
+
+    Args:
+        dataset (CustomDataset): The distilabel dataset to binarize.
+        seed (int, optional): Random seed. Defaults to 42.
+        strategy (BinarizationStrategies, optional): Method to binarize the data. Defaults to "random".
+        keep_ties (bool, optional):
+            Whether to keep ties in case the binarization method generated the chosen
+            and rejected responses to have the same rating. Defaults to True.
+        kwargs: Extra parameters passed to `datasets.Dataset.map`.
+
+    Raises:
+        ValueError: If the strategy is not implemented.
+
+    Returns:
+        CustomDataset: Dataset binarized.
+    """
+    rating_column = "rating"
+    responses_column = "generations"
+
+    def binarize_random(example):
```

Comment:
type hints

Reply:
they are not that usual in internal functions, and that's the standard example from [HuggingFace's datasets](https://huggingface.co/docs/datasets/process#map).